homeassistant/components/tado/device_tracker.py

```diff
@@ -1,149 +1,126 @@
 """Support for Tado Smart device trackers."""
 from __future__ import annotations
 
-import asyncio
-from collections import namedtuple
-from datetime import timedelta
-from http import HTTPStatus
 import logging
+from typing import Any
 
-import aiohttp
-import voluptuous as vol
-
-from homeassistant.components.device_tracker import (
-    DOMAIN,
-    PLATFORM_SCHEMA as BASE_PLATFORM_SCHEMA,
-    DeviceScanner,
-)
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_create_clientsession
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.typing import ConfigType
-from homeassistant.util import Throttle
-
-_LOGGER = logging.getLogger(__name__)
-
-CONF_HOME_ID = "home_id"
-
-MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
-
-PLATFORM_SCHEMA = BASE_PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_HOME_ID): cv.string,
-    }
-)
-
-
-def get_scanner(hass: HomeAssistant, config: ConfigType) -> TadoDeviceScanner | None:
```

Comment:
Implement `async_get_scanner` and start a config flow at the import step. Pass the config as data to the import step.

If the home id is already used in a config entry or if the username and password combination is already used we should abort the config flow. The home id will be set as unique id of the config entry and the username and password can be checked with `_async_abort_entries_match`.

Reply:
We should also create a repair issue asking the user to remove the old config after the import. Also create a repair issue if the import is needed but fails and aborts.

Search for `async_create_issue` to see examples.

Reply:
I think the import step is not needed. The `configuration.yaml` device tracker part is redundant and can be completed removed. After this PR/patch, all the needed configuration is obtained via the TadoConnector. The YAML had an (old way?) way of working where the username, password and home_id were asked again, whilst this was already configured in the initial config flow.
With my current knowledge, I think only an `async_create_issue` is sufficient to warn people to remove the device tracker from their `configuration.yaml`.

Reply:
Alright - I'll work it out. Is my interpretation right that this should be added as an `async_import_step` in the `config_flow.py` file?

Reply:
`async_step_import` yes. Search for existing examples. 👍

Reply:
Kindly asking your review on this part. :)